sandpack-react/src/contexts/theme-context.tsx

```diff
@@ -19,13 +22,16 @@ const SandpackThemeContext = React.createContext<{
 const SandpackThemeProvider: React.FC<{
   theme?: SandpackPredefinedTheme | SandpackPartialTheme;
 }> = props => {
-  const { theme, id } = createThemeObject(props.theme);
+  let themeObj = createThemeObject(
+    getDarkModePreference() ? 'codesandbox-dark' : 'codesandbox-light'
+  );
   const c = useClasser('sp');
-
   // If theme is not explicitly set, don't inject any stylesheet
   if (props.theme) {
-    injectThemeStyleSheet(theme, id);
+    themeObj = createThemeObject(props.theme);
```

Comment:
if you make the changes above, you can keep the injectThemeStyleSheet behind the props.theme check